consumer.go

```diff
@@ -144,10 +144,16 @@ func (c *consumer) ConsumePartition(topic string, partition int32, offset int64)
 		return nil, err
 	}
 
-	var leader *Broker
+	var replica *Broker
 	var err error
-	if leader, err = c.client.Leader(child.topic, child.partition); err != nil {
-		return nil, err
+	if c.conf.Consumer.ReplicaSelector == nil {
+		if replica, err = c.client.Leader(child.topic, child.partition); err != nil {
```

Comment:
I wonder if it is worth setting up ReplicaSelector in NewConfig to be:
```go
func(topic string, partition int32, client Client) (*Broker, error) {
        return client.Leader(child.topic, child.partition)
}
```
and then you wouldn’t need the nil checking logic to choose how to select which partition to consume from in here

Reply:
Good idea. I'll get that in.

Reply:
I have adjusted per your feedback.